http-gateways/PATH_GATEWAY.md

```diff
@@ -0,0 +1,665 @@
+
+The first parameter passed in this header indicates if content should be
+displayed `inline` by the browser, or sent as an `attachment` that opens the
+“Save As” dialog:
+- `Content-Disposition: inline` is the default, returned when request was made
+  with  `download=false`  or a custom `filename` was provided with the request
+  without any explicit `download` parameter.
+- `Content-Disposition: attachment` is returned only when request was made with
+  the explicit  `download=true`
+
+The remainder is an optional `filename` parameter that will be prefilled in the
+“Save As” dialog.
+
+NOTE: when the `filename` includes non-ASCII characters, the header must
+include both ASCII and UTF-8 representations for compatibility with legacy user
+agents and existing web browsers. 
+
+To illustrate, `?filename=testтест.pdf` should produce:
+`Content-Disposition inline; filename="test____.jpg"; filename*=UTF-8''test%D1%82%D0%B5%D1%81%D1%82.jpg`
+  - ASCII representation must have non-ASCII characters replaced with `_`
+  - UTF-8 representation must be wrapped in Percent Encoding ([RFC 3986, Section 2.1](https://www.rfc-editor.org/rfc/rfc3986.html#section-2.1)).
+    - NOTE: `UTF-8''` is not a typo – see [Examples in RFC5987](https://datatracker.ietf.org/doc/html/rfc5987#section-3.2.2)
+
+`Content-Disposition` must be also set when a binary response format was requested:
+
+- `Content-Disposition: attachment; filename="<cid>.car"` should be returned
+  with `Content-Type: application/vnd.ipld.car` responses to ensure client does
+  not attempt to render streamed bytes. CID and `.car` file extension should be
+  used if  a custom `filename` was not provided with the request.
+
+- `Content-Disposition: attachment; filename="<cid>.bin"` should be returned
+  with `Content-Type: application/vnd.ipld.raw` responses to ensure client does
+  not attempt to render raw bytes. CID and `.bin` file extension should be used
+  if  a custom `filename` was not provided with the request.
+
+### `Content-Length` (response header)
+
+Represents the length of returned HTTP payload.
+
+NOTE: the value may differ from the real size of requested data if compression or chunked `Transfer-Encoding` are used.
+<!-- TODO (https://github.com/ipfs/in-web-browsers/issues/194) IPFS clients looking for UnixFS file size should use value from `X-Ipfs-DataSize` instead. -->
+
+### `Content-Range` (response header)
+
+Returned only when request was a [`Range`](#range-request-header) request.
+
+See [RFC7233#header.content-range](https://httpwg.org/specs/rfc7233.html#header.content-range).
+
+### `Accept-Ranges` (response header)
+
+Optional, returned to explicitly indicate if gateway supports partial HTTP
+[`Range`](#range-request-header) requests for a specific resource.
+
+For example, `Accept-Ranges: none` should be returned with
+`application/vnd.ipld.car` responses if the block order in CAR stream is not
+deterministic.
+
+### `Location` (response header)
```

Comment:
Would it make sense to also have a [Content-Location](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Location) header in responses which would contain the canonical `ipfs://{cid}/` URL for the resource?

Reply:
Related to this (more details) https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Location

Reply:
`Content-Location` may not be the header for the job, let's discuss in https://github.com/AgregoreWeb/agregore-browser/issues/192#issuecomment-1168862909